crates/ruff_python_formatter/src/other/match_case.rs

```diff
@@ -33,19 +38,12 @@ impl FormatNodeRule<MatchCase> for FormatMatchCase {
                     &format_with(|f| {
                         write!(f, [text("case"), space()])?;
 
-                        let leading_pattern_comments = comments.leading(pattern);
-                        if !leading_pattern_comments.is_empty() {
```

Comment:
This strikes me as unusual (shouldn't the pattern format its own leading comments?) and the tests seem to pass without it. Hmm...

Reply:
I think the idea was to avoid handling comments for _each_ pattern and handle it at once in the `MatchCase` node. These would only include comments surrounding the pattern itself and not the ones which are in between the pattern (for example, comments between sequence element).